### ISSUE TITLE
lazagne.exe 2.4.3 standalone Chrome passwords

### ISSUE THREAD

**Issue by Papotito123** [AlessandroZ/LaZagne#426]
Hello:
I see a new lazagne 2.4.3 standalone. I try it. But doesn't recover Chrome.
I ran the lazagne.py  2.4.3 and worked well.
I also tried to compile 2.4.3 and did the same ,no Chrome.
My computer installed an update ; 
September 10, 2019-KB4514601 Cumulative Update for .NET Framework 3.5, 4.7.2 and 4.8 for Windows 10 Version 1809 and Windows Server 2019
I don't know if this can cause troubles for compiling to exe.

I test using lazagne.exe 2.3.2 and works well.

I send a txt with print lines added.

Thanks.
[Papotito123_lazagne 2.4.3 standalone_raw cmd.txt](https://github.com/AlessandroZ/LaZagne/files/3619041/Papotito123_lazagne.2.4.3.standalone_raw.cmd.txt)

**Comment by AlessandroZ** [comment]
Aaaaaa finally I managed to reproduce the error. I have no time to fix it but I thing I understood what's going on. I think this occur because the binary that I uploaded is a x86 process architecture whereas your system has an x64 architecture. I think, this difference of arch produce the error. 

I will take a look when I will have more time. 

Thanks for your feedback. :+1:

**Comment by Papotito123** [comment]
Hello:
As always,so much thanks.

**Comment by Papotito123** [comment]
Hello:
I have been uninstalling/reinstalling lazagne 2.4.3(among other versions) ,also updating/upgrading Python 2.7 rsa,psutil ,pyinstaller-develop and others .And has the same results; lazagne.py works good but the compiled lazagne.exe didn't recover Chrome.

But it really recover Chrome encrypted, but seems it can't output/convert strings to plain-text I suspect due to Windows 10 latest version.
Why I say that.?
In my Windows 10 1809 x64 , lazagne.exe 2.4.3 compiled gives the error:
   WindowsError: exception: access violation reading 0x0000045E9D08775A

So,I run the same lazagne.exe 2.4.3 compiled to a Windows 10240 x64.
It runs good and recovered Chrome, or maybe as I see it, it prints output plain-text Chrome password.

[Papotito123_lazagne 2.4.3 Compiled_Win10240 X64_raw cmd.txt](https://github.com/AlessandroZ/LaZagne/files/3629269/Papotito123_lazagne.2.4.3.Compiled_Win10240.X64_raw.cmd.txt)


I hope this helps.

**Comment by Papotito123** [comment]
Hello:
I have been playing with some of your lazagne standalone versions from your page.
Just as info.

I tested your standalones v2.3, v2.3.2, v2.3.2 and all recover Chrome.

I tested your standalones v2.4.2 ,v2.4.3 and both didn't recover Chrome(you already knows).

In other hand, lazagne 2.4(AlessandroZ released this on Sep 2, 2018) standalone and lazagne.py ,both recover Chrome.
But the lazagne 2.4 compiled exe gives the same Chrome error(WindowsError: exception: access violation reading).

So I have a txt with laZagne.py v2.4 raw cmd and lazagne.exe v2.4 compiled.

[Papotito123_lazagne v2.4_raw cmd.txt](https://github.com/AlessandroZ/LaZagne/files/3638380/Papotito123_lazagne.v2.4_raw.cmd.txt)

Thanks.

**Comment by Papotito123** [comment]
Hello:
I have been playing with some of your lazagne standalone versions from your page.
Just as info.

I tested your standalones v2.3, v2.3.2, v2.3.2 and all recover Chrome.

I tested your standalones v2.4.2 ,v2.4.3 and both didn't recover Chrome(you already knows).

In other hand, lazagne 2.4(AlessandroZ released this on Sep 2, 2018) standalone and lazagne.py ,both recover Chrome.
But the lazagne 2.4 compiled exe gives the same Chrome error(WindowsError: exception: access violation reading).

So I have in one txt the laZagne.py v2.4 raw cmd and lazagne.exe v2.4 compiled  raw cmd.

[Papotito123_lazagne v2.4_raw cmd.txt](https://github.com/AlessandroZ/LaZagne/files/3638383/Papotito123_lazagne.v2.4_raw.cmd.txt)

Thanks.

**Comment by Papotito123** [comment]
Hello;
According to your page;

LaZagne v2.3.2 is out
@AlessandroZ AlessandroZ released this on Mar 22, 2018 · 647 commits to master since this release

Windows
Big code review
Lots of minor bug fixed
,and so on.

Note: right now, LaZagne x86 cannot read memory from a x64 process (so some modules cannot work using this build such as wdigest passwords)
That's why, two lazagne binaries have been built (x86 and x64)

I see that some people are having issues with Chrome passwords with your standalones v2.4.2 ,v2.4.3.

In other hand, lazagne 2.4(AlessandroZ released this on Sep 2, 2018) standalone and lazagne.py ,both recover Chrome but throws some errors that can't make output available(but still can see Chrome passwords in the raw cmd).But the compiled exe gives the same Chrome error(WindowsError: exception: access violation reading).
You did something in the 2.4 standalone that is not in the lazagne.py?

This is a raw cmd of .py + standalone + compiled with the prints that hopes be helpful; 
[Papotito123_2.4_py+standalone+compiled_raw cmd.txt](https://github.com/AlessandroZ/LaZagne/files/3674237/Papotito123_2.4_py%2Bstandalone%2Bcompiled_raw.cmd.txt)
 
Thanks in advanced.

**Comment by Papotito123** [comment]
Hello:
Chrome passwords can't be recovered.
This is from lazagne.exe 2.4.2(AlessandroZ released this on Feb 18) downloaded from website.

 ------------------- Google chrome passwords -----------------

[!] Database found: C:\Users\TESTACCOUNT\AppData\Local\Google\Chrome\User Data\Default\Login Data
[!] Temporary db copied: c:\users\testac~1\appdata\local\temp\tbdxfpskt
[!] Traceback (most recent call last):
  File "lazagne\softwares\browsers\chromium_based.py", line 74, in _export_credentials
  File "lazagne\config\winstructure.py", line 542, in Win32CryptUnprotectData
  File "lazagne\config\winstructure.py", line 499, in getData
WindowsError: exception: access violation reading 0x000001C75AB8619D

[!] Traceback (most recent call last):
  File "lazagne\softwares\browsers\chromium_based.py", line 74, in _export_credentials
  File "lazagne\config\winstructure.py", line 542, in Win32CryptUnprotectData
  File "lazagne\config\winstructure.py", line 499, in getData
WindowsError: exception: access violation reading 0x000001C75AB864FA

[!] Traceback (most recent call last):
  File "lazagne\softwares\browsers\chromium_based.py", line 74, in _export_credentials
  File "lazagne\config\winstructure.py", line 542, in Win32CryptUnprotectData
  File "lazagne\config\winstructure.py", line 499, in getData
WindowsError: exception: access violation reading 0x000001C75AB8628A

This issue for recovering Chrome accounts starts to show in your binaries(and the compiled) from lazagne.exe 2.4.2 and up until latest(as of today).

laZagne.py works good in all versions from LaZagne v2.3.2 is out(@AlessandroZ AlessandroZ released this on Mar 22, 2018) until the latest (as of today).

You said ,in this same #issue426,(I think this occur because the binary that I uploaded is a x86 process architecture whereas your system has an x64 architecture. I think, this difference of arch produce the error.).
But,lazagne.exe 2.4 (AlessandroZ released this on Sep 2, 2018) standalone, recovered Chrome but throws some errors that can't make output available(but still I can see Chrome passwords in the raw output in cmd).And this binary is a x86 process architecture.

Just giving some more info for your debugging.

Thanks.

**Comment by Papotito123** [comment]
Hello:
You did some housekeeping in Issues tab.

I tested your standalones v2.3, v2.3.2, v2.3.2 and all recover Chrome passwords.

I tested your lazagne standalones binaries  v2.4.2 ,v2.4.3 and none didn't recover Chrome.Both binary are .x86 process architecture.

Your lazagne.exe 2.4 (AlessandroZ released this on Sep 2, 2018) standalone binary can recovered Chrome but throws some errors that can't make the output log available(but still I can see Chrome passwords fetched in the cmd when running -vvv).Also this binary is a x86 process architecture.

WindowsError: exception: access violation reading 0x000001C75AB8628A

This issue for recovering Chrome accounts starts to show in your standalone binaries(and in the compiled .exe) from lazagne v2.4.2 and up until latest version(as of today).

Your last lazagne binary that came with 2 separated .exe,1 for x86 and 1 for x64,was lazagne v2.3.2 because as you said,LaZagne x86 cannot read memory from a x64 process (so some modules cannot work using this build such as wdigest passwords).So that's why you released 2 binaries.

Then you released lazagne v2.4 standalone binary that indeed fetch and show Chrome accounts in the cmd window ,but throws some errors that doesn't let to finish and due to this sudden termination  the output log is not written.
Also I re-run and verify some dependencies and still Chrome issue.I'm still using python 2.7.16.

I hope this time I was more clear in what really want to explain.
And be of help .

Thanks.

**Comment by AlessandroZ** [comment]
Since 2.4, the binary is compiled automatically using travis in x64 bit. So now, to compile it in x86 bit, you should install a x86 python interpreter and use it to compile it. It should work. 

It's much more easier for me to use travis than to compile it manually. 

The error is not on the code but on the process arch. Try to compile manually an x86 process, it should work.

**Comment by Papotito123** [comment]
Hello:
I had some thoughts about installing also python x86 but was afraid.So ,if I install python x86 ,do I have to delete the python x64?
I will try it and keep in touch.

Thanks for your explanation about your binaries.

**Comment by Papotito123** [comment]
Hello:
I managed to install Python x86.And see how important paths can be and other things,
In the end ,I have to rename the original Python27 folder( the x64 installation) to other and rename the python x86 install to Python27.Now I saw how when running, pyinstaller.py --onefile -w "folder-path-to/lazagne.spec", will use WinSxS x86_Microsoft.VC90.CR needed files, instead of WinSxS amd64.
So then I see what is needed to create the healthy lazagne.exe , Bootloader C:\pyinstaller-develop\PyInstaller\bootloader\Windows-32bit\run.exe .

It recover Chrome,hash are goods,recover Firefox. 
I am using the latest laZagne.py from https://github.com/AlessandroZ/LaZagne.
running this laZagne.py in Python27 x64 recovers well the WIFI password.

But running this new compiled x86 with -vvv ,shows this error(first time I see this error);
 ------------------- Wifi passwords -----------------

[!] Trying using netsh method
[-] Traceback (most recent call last):
  File "lazagne\softwares\wifi\wifi.py", line 84, in run
  File "lazagne\softwares\wifi\wifi.py", line 40, in decrypt_using_netsh
  File "lazagne\softwares\wifi\wifi.py", line 40, in <genexpr>
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd1 in position 0: ordinal not in range(128)

[-] Password not found !!!
INFO: [!] Password not found.
Authentication: WPA2PSK
Protected: true
SSID: CLAROXXXXXX

 ------------------- Memory_dump passwords -----------------

Then,WIFI password is not recovered.

So much thanks.
